### DISCUSSION TITLE
Adding header to the curl is causing issues

### DISCUSSION THREAD

**Discussion by Muhammadh1994** [istio/istio#53579] [Q&A]
I have a curl
```
example.net/appname --header 'host: appname-env'
```
In Nginx ingress I just had two hosts withing the same ingress rule and it worked.
Currently I am migrating to istio.
In hosts I have both  example.net and appname-env
But when I curl I keep getting 404 from the envoy but It works fine if I remove the  --header 'host: appname-env'
Any help on this would be appreciated

**Comment by howardjohn** [comment]
This is probably not going to be full answers without more information about your configuration. But Istio does most routing based on hostname so those are very very different requests from Istio perspective
